### PR TITLE
Add `$schema` to `cgmanifest.json`

### PR DESCRIPTION
**Pull request by JamieMagee**
This pull request adds the JSON schema for `cgmanifest.json`.

### Why?

A JSON schema helps you to ensure that your `cgmanifest.json` file is valid.
JSON schema validation is a built-in feature in most modern IDEs like Visual Studio and Visual Studio Code.
Most modern IDEs also provide code-completion for JSON schemas.

### How can I validate my `cgmanifest.json` file?

Most modern IDEs like Visual Studio and Visual Studio Code have a built-in feature to validate JSON files.
You can also use [this small script](https://github.com/JamieMagee/verify-cgmanifest) to validate your `cgmanifest.json` file.

### Why does it suggest camel case for the properties?

Component Detection is able to read camel case and pascal case properties.
However, the JSON schema doesn't have a case-insensitive mode.
We therefore suggest camel case as it's the most common format for JSON.

### Why is the diff so large?

To deserialize the `cgmanifest.json` file, we use [`JSON.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse).
However, to serialize the JSON again we use [`prettier`](https://prettier.io/).
We found that, in general, it gave smaller diffs than the default [`JSON.stringify()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) function.

### CODE REVIEW

**Review by kuzminrobin — APPROVED** [review]

File: src/Qir/Common/Externals/cgmanifest.json [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-    ]
-}
+      }
+    },
+    {
+      "Component": {
+        "Type": "git",
+        "Git": {
+          "RepositoryUrl": "https://github.com/catchorg/Catch2",
+          "CommitHash": "87074da73ecb1c2e7c35fd14f50ca21c4c002adc"
```
[the remark this answers is not in the document]
Comment (JamieMagee) [reply src/Qir/Common/Externals/cgmanifest.json:19]: False positive. This is a git commit hash.

File: src/Qir/Common/Externals/cgmanifest.json [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+        }
+      }
+    },
+    {
+      "Component": {
+        "Type": "git",
+        "Git": {
+          "RepositoryUrl": "https://github.com/CLIUtils/CLI11",
+          "CommitHash": "5cb3efabce007c3a0230e4cc2e27da491c646b6c"
```
[the remark this answers is not in the document]
Comment (JamieMagee) [reply src/Qir/Common/Externals/cgmanifest.json:28]: False positive. This is a git commit hash.

**Comment by JamieMagee** [comment]
@kuzminrobin could you merge this please, if you're happy with the changes? I don't have permissions in this repository.

**Comment by kuzminrobin** [comment]
@JamieMagee, 
"This branch is out-of-date with the base branch". You need to either [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) the current branch on top of latest main or merge main to the current branch.
The "Code scanning results / devskim" pipeline fails. Please resolve the failures (likely by pressing the button "Dismiss alert"; seethe picture below).
image

**Comment by JamieMagee** [comment]
@kuzminrobin I've updated the branch, but I do not have permissions to dismiss the alert.

**Comment by kuzminrobin** [comment]
@JamieMagee, please let me know when the build succeeds.

**Comment by kuzminrobin** [comment]
Merged to main. Feel free to remove your branch.

**Pull request merged**